File: sdk-cluster-jakarta-ws/src/main/java/dev/getelements/elements/sdk/jakarta/ws/JakartaWebsocketRemoteInvoker.java
package dev.getelements.elements.sdk.jakarta.ws;

import dev.getelements.elements.sdk.cluster.remote.*;
import jakarta.websocket.ClientEndpoint;

import java.util.List;
import java.util.concurrent.CompletionStage;
import java.util.concurrent.TimeUnit;
import java.util.function.Consumer;

@ClientEndpoint
public class JakartaWebsocketRemoteInvoker implements RemoteInvoker {

    @Override
    public String getConnectAddress() {
        return "";
    }

    @Override
    public void start(String connectAddress, long timeout, TimeUnit timeoutTimeUnit) {

    }

    @Override
    public AsyncOperation invokeAsync(Invocation invocation, List<Consumer<InvocationResult>> asyncInvocationResultConsumerList, InvocationErrorConsumer asyncInvocationErrorConsumer) {
        return null;
    }

    @Override
    public CompletionStage<Object> invokeCompletionStage(Invocation invocation, List<Consumer<InvocationResult>> asyncInvocationResultConsumerList, InvocationErrorConsumer asyncInvocationErrorConsumer) {
        return null;
    }

}

File: sdk-cluster-jakarta-ws/src/main/kotlin/dev/getelements/elements/sdk/cluster/jakarta/ws/JakartaWebsocketRemoteInvoker.kt
package dev.getelements.elements.sdk.cluster.jakarta.ws

import dev.getelements.elements.sdk.cluster.remote.AsyncOperation
import dev.getelements.elements.sdk.cluster.remote.Invocation
import dev.getelements.elements.sdk.cluster.remote.InvocationErrorConsumer
import dev.getelements.elements.sdk.cluster.remote.InvocationResult
import dev.getelements.elements.sdk.cluster.remote.RemoteInvoker
import dev.getelements.elements.sdk.util.io.PayloadReader
import dev.getelements.elements.sdk.util.io.PayloadWriter
import java.util.concurrent.CompletionStage
import java.util.function.Consumer

class JakartaWebsocketRemoteInvoker(
    private val connectAddress: String,
    private val payloadReader: PayloadReader,
    private val payloadWriter: PayloadWriter
) : RemoteInvoker {

    override fun invokeAsync(
        invocation: Invocation,
        asyncInvocationResultConsumerList: List<Consumer<InvocationResult>>,
        asyncInvocationErrorConsumer: InvocationErrorConsumer
    ): AsyncOperation? = null

    override fun invokeCompletionStage(
        invocation: Invocation,
        asyncInvocationResultConsumerList: List<Consumer<InvocationResult>>,
        asyncInvocationErrorConsumer: InvocationErrorConsumer
    ): CompletionStage<Any> = JakartaWebsocketRemoteInvocation(
        invocation,
        asyncInvocationResultConsumerList,
        asyncInvocationErrorConsumer).send();

}
